File: manual-test.py
# Manual test file for AgentWatch
import os
import subprocess

def execute_command(user_command):
    # SECURITY ISSUE: Command injection vulnerability
    result = subprocess.run(user_command, shell=True, capture_output=True, text=True)
    return result.stdout

def process_file(filename):
    # SECURITY ISSUE: Path traversal vulnerability
    file_path = "/tmp/" + filename
    try:
        with open(file_path, 'r') as f:
            content = f.read()
        return content
    except FileNotFoundError:
        return "File not found"

def authenticate_user(username, password):
    # SECURITY ISSUE: Hardcoded credentials
    admin_users = {
        "admin": "password123",
        "root": "admin"
    }
    
    if username in admin_users and admin_users[username] == password:
        return {"authenticated": True, "role": "admin"}
    
    return {"authenticated": False, "role": "guest"}

# Main execution
if __name__ == "__main__":
    print("Manual test file ready for AgentWatch analysis!")
    
    # Test functions
    result = execute_command("echo 'Hello World'")
    print(f"Command result: {result}")

File: manual-test.js
// Fresh manual test for AgentWatch manual-only mode
function authenticateUser(credentials) {
  // Security issue: hardcoded admin credentials
  if (credentials.username === "admin" && credentials.password === "admin123") {
    return { authenticated: true, role: "admin" };
  }
  
  // Weak password validation
  if (credentials.password.length < 6) {
    throw new Error("Password too short");
  }
  
  return { authenticated: false };
}

function sanitizeInput(userInput) {
  // XSS vulnerability - no actual sanitization
  return userInput.replace(/<script>/gi, "");
}

function logUserAction(action, userId) {
  // Potential information disclosure
  console.log(`User ${userId} performed action: ${action} at ${new Date()}`);
}

// Export functions for testing
module.exports = {
  authenticateUser,
  sanitizeInput,
  logUserAction
};